feat(arc): Architectural Review Committee (#3)
gccloudone/aurora: content/en/technical-advisory-group/architectural-review-committee/ARC-001.en.md

lang: "en"
---

**Date:** March 26th, 2025
**Location:** [Virtual/In-Person]  
**Presenter:** William Hearn (Senior Cloud Architect, GC Cloud One)  
**Attendees:** ARC Members
**Objective:** Review the Aurora initiative and secure approval for its **strategic direction and technical architecture**.

---

## 1. Executive Summary

### **Context**

> The **Aurora Initiative** is a **unified application hosting platform** designed to standardize **managed Kubernetes cloud offerings** across SSC. It integrates **both proprietary (OpenShift) and OSS (Cloud Native Kubernetes) environments**, drawing from operational experience in the **Cloud Native Platform (CNP)**.

Reviewer: ```suggestion
The **Aurora Initiative** is a **unified application hosting platform** designed to standardize **managed Kubernetes cloud offerings** across the GC. It integrates **both proprietary (OpenShift) and OSS (Cloud Native Kubernetes) environments**, drawing from operational experience in the **Cloud Native Platform (CNP)**.
```

> The **Aurora Initiative** is a **unified application hosting platform** designed to standardize **managed Kubernetes cloud offerings** across SSC. It integrates **both proprietary (OpenShift) and OSS (Cloud Native Kubernetes) environments**, drawing from operational experience in the **Cloud Native Platform (CNP)**.

Reviewer: ```suggestion
The **Aurora Initiative** is a **unified application hosting platform** designed to standardize **managed Kubernetes cloud offerings** across the GC. It integrates **both proprietary (OpenShift) and OSS (Cloud Native Kubernetes) environments**, drawing from operational experience in the **Cloud Native Platform (CNP)**.
```

### **Why Now?**

> - Current departmental cloud-native strategies are **disjointed, costly, and redundant**.

Reviewer: You could also talk about the move to centralized cloud at SSC, and the new GC Cloud One initiative.

- Aurora provides a **governance framework** that reduces inefficiencies, promotes **multi-cloud standardization**, and aligns with SSC’s **GC Cloud One** strategy.
- The initiative has **received broad buy-in** from multiple departments and aligns with similar government initiatives (BC Gov DevOps, Platform One, etc.).

---

## 2. Architecture Overview

> ### **2.1 Technical Architecture**

Reviewer: It might be good to emphasize that Aurora is a set of add-on components that bring a full-featured app platform to a Kubernetes cluster. So it uses the cloud Kubernetes offerings as a foundation and builds on-top of it.

- **Core Services:** Kubernetes (AKS, EKS, GKE, OpenShift)
- **Networking:** Cilium + BGP (BIRD with route reflectors)
- **Security Model:** Zero Trust, Gatekeeper (OPA), SCED for on-prem connectivity
- **CI/CD:** GitLab Runners, Crossplane for cloud resource automation
- **Observability:** Loki, Prometheus
- **Multi-Tenancy Approach:** Namespaced isolation, policy enforcement

#### **Key Benefits**

> ✔️ **Unified API surface** for Kubernetes adoption across SSC.  

Reviewer: ```suggestion
✔️ **Unified API surface** for Kubernetes adoption across the GC.  
```

You can also touch on how it will "abstract" the implementation differences between the clouds, making it seamless for workloads* to move between cloud providers.

✔️ **Flexible deployment models** (OSS and proprietary).  

> ✔️ **Security-first** approach (zero trust, policy-driven enforcement).  

Reviewer: You can touch on it being an opinionated configuration of the cloud components (such as AKS) and the Aurora components that aligns with GC requirements. Namespace-isolation (Kubernetes resources) is supported by namespace network isolation through network policies, providing a stronger multi-tenant model.

For policy-driven enforcement.. the intention is to empower workload owners to do all activities themselves while being bounded by policy to ensure they cannot violate GC policy.

✔️ **Interoperability** with existing departmental solutions.

---

> ## 3. Governance & Compliance Alignment

Reviewer: I think this is somewhere to bring up the TAG groups, and that the community is involved in the building of Aurora. This ensures that the Aurora product meets the needs of its users, and solves problems that bring the greatest return in investment.

- Aurora **adheres to SSC’s cloud governance model**, ensuring **interoperability, security, and standardization**.
- Leverages **CNCF best practices** for operational maturity.
- Supports **multi-tenancy, workload separation, and automated compliance checks**.
- Aligns with **Treasury Board Secretariat (TBS) cloud directives**.

### **Key Decision Point for ARC**

✔️ Does Aurora meet SSC’s **interoperability and security requirements**?

---

## 4. Operational Model

### **4.1 Deployment Approach**

Aurora will be deployed in SSC’s **Enterprise Tenant**, with:

> - **Two main environments:** `AuroraNonProd` (mgmt-dev-cc-00) and `AuroraProd` (mgmt-prod-cc-00).

Reviewer: I think here is where you need to talk about this at a wider view. Specifically:

- There will be two "development services" environments (non-prod and prod) for running services that support the Aurora environments and its users. Currently targeted include: Artifactory, X-Ray, Backstage, etc.
- [Planned] A shared management environment to run services that manage and operate all the Aurora clusters (Argo CD). *Problems to be solved: how to ensure connectivity from this cluster to all clusters operating across the Enterprise tenant and Department tenants/landing zones.
- [Planned] Shared SDLC environments in the Enterprise tenant that can be consumed by small departments/agencies
- [Planned] Per-department SDLC clusters deployed in (ideally) the Enterprise Tenant or (alternatively) the departmental tenant, but connected to Departmental Landing Zones.

> - Integration with **Fortigate Firewalls** for automated rule management.

Reviewer: ```suggestion
- Integration with **Fortigate Firewalls** for metadata-based rulesets (limiting connectivity to the appropriate workload resources within shared clusters) managed via IaC
```

- Centralized **GitOps pipeline** (Argo CD) for policy-based application deployment.

### **4.2 Team Structure**

- **Architecture & Governance Team**: Defines standards and compliance.
- **Platform Operations Team**: Maintains core services (GitLab, Vault, Artifactory).

> - **Advisory SIGs (Architecture, Networking, Security)**: Drives technical decisions.

Reviewer: ```suggestion
- **Advisory SIGs (Architecture, Networking, Security)**: Drives technical decisions.
```

I think this will need to switch to TAG instead of SIG

Aurora includes a curated, opinionated configuration of cloud-native components and Kubernetes resources aligned to Government of Canada requirements—balancing developer self-service with strong security, compliance, and operational guardrails.

- **Base Platform:** Cloud-managed Kubernetes offerings (AKS, EKS, GKE, OpenShift)
- **Networking Layer:** Cilium with BGP (BIRD + route reflectors) and namespace-level network policies for tenant isolation
- **Security Model:** Zero Trust architecture, Gatekeeper (OPA) for policy enforcement, SCED integration for on-prem connectivity
- **CI/CD Pipelines:** GitLab Runners integrated with Argo CD for GitOps-based delivery
- **Observability Stack:** Prometheus, Loki, and centralized logging/metrics forwarding

> - **Multi-Tenancy Approach:** Namespaced resource boundaries, enforced network isolation, and RBAC + policy guardrails

Reviewer: It might be worth mentioning that in API isolation will be looked at down the road, to provide a mechanism for those wishing to build or operate advanced services can do so without impacting other workloads (e.g., using CRDs)

- Aurora **adheres to SSC’s cloud governance model**, ensuring **interoperability, security, and standardization**
- Leverages **CNCF best practices** for operational maturity
- Supports **multi-tenancy, workload separation, and automated compliance checks**
- Aligns with **Treasury Board Secretariat (TBS) cloud directives**

### Key Decision Point for ARC

✔️ Does Aurora meet SSC’s **interoperability and security requirements**?
✔️ Is the **Technical Advisory Group (TAG)** recognized as a valid mechanism for cross-departmental technical governance?

---

## 4. Operational Model

Aurora’s operational model reflects a scalable, multi-environment architecture grounded in platform engineering principles. It is designed to centralize operational overhead, reduce duplication across departments, and enable secure, self-service cloud-native development at scale. The model supports both shared services and per-department configurations, enabling a flexible yet governed path for onboarding departments of all sizes.

### 4.1 Deployment Approach

Aurora will be deployed in SSC’s **Enterprise Tenant**, with:

- **Two main platform environments** for cluster provisioning and platform services:

>   - `AuroraNonProd` (`mgmt-dev-cc-00`)

Reviewer: Might be good to mention/elaborate that the use of this environment is to test the cluster provisioning / platform services before they are promoted to the production environment, where all the actual clusters will be managed from.

Just helps get ahead of questions of what the mgmt-dev vs. mgmt-prod manages.

Reviewer: Another point too, these bullet points seem somewhat redundant with the points in 4.2. Might be good to try and merge them together to avoid the redundancy.

- **Advisory TAGs** (*Architecture*, *Networking*, *Security*): Serve as expert panels to guide technical decisions, validate designs, and drive community alignment

### Key Decision Point for ARC

✔️ Does this operational model align with SSC’s cloud-native strategy?
✔️ Does the structure support long-term scalability across departments and tenants?

---

## 5. Risks & Mitigations

| **Risk**                           | **Mitigation Strategy**                                                                           |
| ---------------------------------- | ------------------------------------------------------------------------------------------------- |
| Departments resist standardization | Aurora allows **flexible adoption** while enforcing key governance policies.                      |
| OSS vs Proprietary Tensions        | Supports **both managed OpenShift and Kubernetes**, providing departments with options.           |
| Security & Compliance Overhead     | **Automated policy enforcement** via Gatekeeper, OPA, and workload isolation.                     |

> | Multi-cloud Complexity             | **Standardized onboarding, automated provisioning**, and governance frameworks reduce complexity. |

Reviewer: I think another mitigation to your multi-cloud strategy is having experts of the different cloud providers in GC Cloud One assigned to working with Aurora for development within that cloud environment.
